tests/output/pjs/output_test.js

```diff
@@ -637,11 +637,24 @@ describe("Scratchpad Output Exec", function() {
     runTest({
         title: "Program functions exist",
         code: function() {
-            var x = Program.settings();
+            debug("start");
+            Program.settings();
             Program.restart();
             Program.runTests(function() {});
         },
-        errors: []
+        errors: [],
+        setup: function(output) {
+            var p = output.output.canvas;
+            sinon.stub(p.Program, "settings");
+            sinon.stub(p.Program, "restart");
+            sinon.stub(p.Program, "runTests");
```

Comment:
processing.Program methods weren't being stubbed and where actually causing certain test code to be run in an infinite loop after ```it``` had completed.